pkg/kratos/handlers.go

```diff
@@ -280,20 +283,39 @@ func (a *API) handleCreateRecoveryFlow(w http.ResponseWriter, r *http.Request) {
 func (a *API) handleGetSettingsFlow(w http.ResponseWriter, r *http.Request) {
 	q := r.URL.Query()
 
-	flow, cookies, err := a.service.GetSettingsFlow(context.Background(), q.Get("id"), r.Cookies())
+	flow, response, err := a.service.GetSettingsFlow(context.Background(), q.Get("id"), r.Cookies())
 	if err != nil {
 		a.logger.Errorf("Error when getting settings flow: %v\n", err)
 		http.Error(w, "Failed to get settings flow", http.StatusInternalServerError)
 		return
 	}
 
+	// If aal1, redirect to complete second factor auth
+	if response.StatusCode == http.StatusForbidden {
+		redirectResp := new(ErrorBrowserLocationChangeRequired)
+		err = unmarshalByteJson(response.Body, redirectResp)
+		if err != nil {
+			a.logger.Errorf("Failed to unmarshal JSON: %s", err)
+			return
+		}
+
+		if redirectResp.HasRedirectTo() {
+			a.logger.Errorf("Error when getting settings flow: %v\n", response)
+			w.WriteHeader(http.StatusInternalServerError)
```

Comment:
Why is this a `500`? AFAICT the request was processed fine, but the user session is not ready. Shouldn't it be 4XX?